zerver/views/muting.py

```diff
@@ -85,3 +86,29 @@ def update_muted_topic(
             stream_name=stream,
             topic_name=topic,
         )
+
+
+def mute_user(request: HttpRequest, user_profile: UserProfile, muted_user_id: int) -> HttpResponse:
+    if user_profile.id == muted_user_id:
+        return json_error(_("Cannot mute self"))
+
+    muted_user_profile = get_user_profile_by_id(muted_user_id)
```

Comment:
This should be `access_user_by_id`, same for the other endpoint -- otherwise you can access a random user ID not in your realm.

Reply:
Done. Changed to `access_user_by_id`.